src/CouponCode.php

```diff
@@ -233,22 +233,24 @@ protected function _normalize($string, array $options = []) {
 		return $string;
 	}
 
-	/**
-	 * Generates a cryptographically secure sequence of bytes.
-	 *
-	 * @param integer $bytes Number of bytes to return.
-	 * @return string
-	 */
+    /**
+     * Generates a cryptographically secure sequence of bytes.
+     *
+     * @param integer $bytes Number of bytes to return.
+     * @return string
+     * @throws Exception
+     */
 	protected function _random($bytes) {
-		if (is_readable('/dev/urandom')) {
```

Comment:
If you run into openbasedir issues with this line, than your openbasedir configuration needs to be changed as it is too strict. It is safe to generally include `/dev/urandom`.